test/modules/http/compliance/http.test.ts

```diff
@@ -163,3 +163,40 @@ it('mocks response to a non-existing host', async () => {
   expect(await text()).toBe('howdy, john')
   expect(requestListener).toHaveBeenCalledTimes(1)
 })
+
+it('returns mocked socket address', async () => {
+  interceptor.on('request', async ({ request }) => {
+    request.respondWith(new Response())
+  })
+
+  const connectPromise = new DeferredPromise<object>()
+  const request = http.get('http://example.com')
+  request.once('socket', socket => {
+    socket.once('connect', () => {
+      connectPromise.resolve(socket.address())
```

Comment:
It would be great to also have a test for `socket.address()` _before_ the `connect` is emitted. I believe it should still be set?

Reply:
It seems like the lookup happens after the socket event is emitted.
```js
request.once('socket', socket => {
  console.log(socket.address()); // prints empty object {}
});
```

We probably need to cover the `lookup` event as well.

Reply:
Yes, this is correct. ClientRequest emits `socket` as soon as the socket is assigned, then the socket goes through its life-cycle, emitting `lookup` and `connect`.